defaults/main.yml

```diff
@@ -53,6 +53,7 @@ vault_datacenter: dc1
 vault_log_level: "{{ lookup('env','VAULT_LOG_LEVEL') | default('info', true) }}"
 vault_iface: "{{ lookup('env','VAULT_IFACE') | default(ansible_default_ipv4.interface, true) }}"
 vault_address: "{{ hostvars[inventory_hostname]['ansible_'+vault_iface]['ipv4']['address'] }}"
+vault_hostname: "{{ vault_address }}"
```

Comment:
Why not like this?
```
vault_hostname: "{{ inventory_hostname }}"
```

Reply:
because this still has to work even if you aren't using a hostname, and for backward compatibility

Reply:
Would this work?

`vault_hostname: "{{ ansible_fqdn|vault_address }}"`

Reply:
So, ansible_fqdn is always assigned via Ansible regardless of being resolvable. So it would be meaningless to have vault_address there. It also would break existing deployments using this role as it would change where the endpoint is expected. If a host has multiple IP addresses and you're hosting this one one of the IP's assuming ansible_fqdn can cause the wrong IP to be returned, ending up in a failed check, even if the service is running.

lets say my ansible_fqdn is ericsysmin, which resolves to 127.0.0.1 but my Vault service is running on 10.0.0.120, and it uses a different domain. How would you handle this?

Prior to this PR we were already using the IP address, so we need to ensure that existing deployments don't get broken because of DNS resolving to different addresses other than the vault_address which could have been set or assigned to the vault server.

Reply:
However, you did just help me realize a problem on checking when using 0.0.0.0 which uses all IP's on the host

Reply:
Removed this from default/main.yml